### ISSUE TITLE
Snapshot Voting Modal unmounted when screen size changes

### ISSUE THREAD

**Issue by alexyoung23j** [hicommonwealth/commonwealth#3573]

**Comment by jnaviask** [comment]
This appears to be all of our modals now, not just Snapshot Voting.

**Comment by alexyoung23j** [comment]
Problem is limited to modals that are rendered by children of <CWContentPage/> components. Investigating.

**Comment by alexyoung23j** [comment]
Diagnosis: 

The CWContentPage component, which is responsible for implementing the responsive design on the thread view page, reacts to changes in screen size and effectively mounts/dismounts its children into different containers when the size changes. This pattern was likely copied from the original mithril implementation and makes sense.

However, it presents a problem when it comes to modals, because unlike with the old modal store system we used, our current implementation renders the Modal component directly from the component that needs it (i.e. controlled with a `modalOpen` state variable). So when the screen size changes and the child components of CWContentPage are rerendered, the modal state is lost. 

Curious to get @masvelio eyes on an approach to this. In theory we can lift the modal components all the way up the tree to CWContentPage's level, but this is a fairly serious refactor and hard to extend down the road.

**Comment by masvelio** [comment]
yeah, diagnosis is right on. From an architectural point of view, the current modal system is different from the one we had in mithril. Over there, modals were controlled by a global state (there was one container that displayed different content). Now, modals are managed and rendered by local states and components. As you correctly pointed out, if a component is unmounted, all its children (along with the modal) are removed from the DOM.

I don't know how to reproduce the error in Snapshot Voting but I wonder if this is generally an edge case? Why would a user want to do a resize window while displaying a modal? I am not saying we should not care about this but maybe the issue is not so severe to do huge refactor in the code. But maybe I don't fully understand the issue.

Changing this approach to similar we had in mithril seems to be a very big lift, especially, that we do not want to use current implementation of global state and we did not figure out new version of global state. I think it would be good to identify where exactly (which modal) the functionality cannot be used because of this architectural pattern, and then, figure out some solution for those specific modals (eg hide component instead of unmounting it or hoist modal up in the tree).

**Comment by kurtisassad** [comment]
I have come up with the following terminology to simplify things. For each modal we have:
* Display Part - Can open the modal.
* Modal Part - The modal itself, can close itself.

For diagrams:
* Squares represent component logic
* Circles represent components

Ok, so I have attempted to hoist the modals to the CWContentPage level, but it seems like a huge refactor, because each modal keeps track of its own state, and this state is tightly coupled to the display part modal itself.

I have another potential solution (although its a little messy). So here is our current DOM/VDOM:
<img width="732" alt="Screenshot 2023-05-05 at 10 35 18 AM" src="https://user-images.githubusercontent.com/14794654/236527333-a3fa5435-c739-4e70-8ec4-af2362af1395.png">

The issue is, as stated above, refactoring the modal part from the child component into the CWContentPage component takes huge refactoring because of its coupling with the display part.

The potential solution involves React Portals API. This API basically allows you to link parts of the DOM manually without needing to alter the VDOM. So if we were to go ahead with this React Portals API, it would look like:

VDOM:
<img width="654" alt="Screenshot 2023-05-05 at 10 41 38 AM" src="https://user-images.githubusercontent.com/14794654/236528574-52d54971-8a1c-499a-a155-9994cb427973.png">

DOM:
<img width="474" alt="Screenshot 2023-05-05 at 10 42 27 AM" src="https://user-images.githubusercontent.com/14794654/236528737-573337eb-615d-4343-9cdd-8b81729e0ebf.png">

With this solution we should be able to solve the issue. The only problem is that this is not really clean, since IMO decoupling the VDOM from the DOM makes the code much more difficult to reason with.

I am not sure this will work in practice, but in theory it seems good. Does anyone have any comments or suggestions on going this route?

**Comment by kurtisassad** [comment]
Labelled 3 points if we go with the portal approach due to current issues that need to be resolved. If we go with a full refactor, it will obviously be much higher.
